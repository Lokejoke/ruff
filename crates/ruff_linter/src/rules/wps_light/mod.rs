//! Rules from [wps-light](https://pypi.org/project/wps-light/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::ListMultiplication, Path::new("WPS435.py"))]
    #[test_case(Rule::ReservedArgumentAsVariable, Path::new("WPS117.py"))]
    #[test_case(Rule::UnderscoresInNumber, Path::new("WPS303.py"))]
    #[test_case(Rule::AssignmentToSubscriptSlice, Path::new("WPS362.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("wps_light").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
