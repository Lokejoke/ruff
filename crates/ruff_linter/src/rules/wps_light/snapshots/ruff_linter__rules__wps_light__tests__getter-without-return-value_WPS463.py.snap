---
source: crates/ruff_linter/src/rules/wps_light/mod.rs
snapshot_kind: text
---
WPS463.py:20:1: WPS463 Getter function returns no value.
   |
18 |           return none_val
19 |   
20 | / def get_val():  # [WPS463]
21 | |     return
   | |__________^ WPS463
22 |   
23 |   def get_val():
   |

WPS463.py:33:5: WPS463 Getter function returns no value.
   |
32 |   class X:
33 |       def get_val():  # [WPS463]
   |  _____^
34 | |         return
   | |______________^ WPS463
35 |   
36 |   def get_val(cond_val):
   |

WPS463.py:46:1: WPS463 Getter function returns no value.
   |
44 |           return None
45 |       
46 | / def get_val(cond_val):  # [WPS463]
47 | |     if cond_val:
48 | |         return
49 | |     while not cond_val:
50 | |         print(cond_val)
   | |_______________________^ WPS463
51 |   
52 |   def get_val(cond_val):
   |
